src/types.ts

```diff
@@ -116,8 +115,15 @@ export interface Features {
  *
  * Implementation for resolving all the required flag types must be defined.
  */
-export interface Provider extends Pick<Partial<EvaluationLifeCycle>, 'hooks'> {
+export interface Provider {
   readonly metadata: ProviderMetadata;
+  /**
+   * A provider hook exposes a mechanism for provider authors to register hooks
+   * to tap into various stages of the flag evaluation lifecycle. These hooks can
+   * be used to perform side effects and mutate the context for purposes of the
+   * provider. Provider hooks are not configured or controlled by the application author.
+   */
+  readonly hooks?: Hook[];
```

Comment:
Either work, but is there any reason why this isn't following the same pattern of `getHooks`?